compiler/noirc_errors/src/position.rs

```diff
@@ -51,27 +81,49 @@ impl<T> std::borrow::Borrow<T> for Spanned<T> {
     }
 }
 
+#[derive(
+    Copy, Clone, Serialize, Eq, PartialEq, Ord, PartialOrd, Debug, Deserialize, Hash, Default,
```

Comment:
`Default` seems like it could possibly introduce bugs to me. I'm unsure if `Default` would always point to src/main.rs or a random file that happens to be first but I don't think we should need a constructor to return either of those two cases.

Reply:
I'm fine removing defaults but this leads to other refactorings, eg
[unwrap_or_default](https://github.com/noir-lang/noir/blob/1d177d2ac8eb1400718a35203fd04d8134e352c1/compiler/noirc_frontend/src/hir/def_collector/errors.rs#L251)
...where at the end of the day we need to provide some kind of default or be more strictier if not definiing source of something is ok to do.

Reply:
Lets just table this then for now I suppose. This PR is already quite large so we can leave any further refactorings for later.